src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -1453,12 +1572,159 @@ def _get_options(throughput=None, max_throughput=None):
+                                    databases_to_restore=databases_to_restore,
+                                    arm_location=target_restorable_account.location)
+
+
+def _convert_to_utc_timestamp(timestamp_string):
+    import dateutil
+    import datetime
+    import dateutil.parser
+    timestamp_datetime = dateutil.parser.parse(timestamp_string)
+    timestamp_datetime_utc = None
+    # Convert to utc only if timezone aware
+    if timestamp_datetime.tzinfo is not None and timestamp_datetime.tzinfo.utcoffset(timestamp_datetime) is not None:
+        timestamp_datetime_utc = timestamp_datetime.astimezone(datetime.timezone.utc)
+    else:
+        timestamp_datetime_utc = timestamp_datetime
+    return timestamp_datetime_utc
+
+
+def cli_cosmosdb_restorable_database_account_list(client,
+                                                  location=None,
+                                                  account_name=None):
+    restorable_database_accounts = None
+    if location is not None:
+        restorable_database_accounts = client.list_by_location(location)
+    else:
+        restorable_database_accounts = client.list()
+
+    if account_name is None:
+        return restorable_database_accounts
+
+    matching_restorable_accounts = []
+    restorable_database_accounts_list = list(restorable_database_accounts)
+    for account in restorable_database_accounts_list:
+        if account.account_name == account_name:
+            matching_restorable_accounts.append(account)
+    return matching_restorable_accounts
+
+
+def cli_retrieve_latest_backup_time(client,
+                                    resource_group_name,
+                                    account_name,
+                                    database_name,
+                                    container_name,
+                                    location):
+    try:
+        client.get_sql_database(resource_group_name, account_name, database_name)
+    except ResourceNotFoundError:
+        raise CLIError("Cannot find a database with name {}".format(database_name))
+
+    try:
+        client.get_sql_container(resource_group_name, account_name, database_name, container_name)
+    except ResourceNotFoundError:
+        raise CLIError("Cannot find a container with name {} under database {}".format(container_name, database_name))
+
+    restoreLocation = ContinuousBackupRestoreLocation(
+        location=location
+    )
+
+    asyc_backupInfo = client.begin_retrieve_continuous_backup_information(resource_group_name,
+                                                                          account_name,
+                                                                          database_name,
+                                                                          container_name,
+                                                                          restoreLocation)
+    return asyc_backupInfo.result()
```

Comment:
Is it best practice to handle LRO by `.result()` within `custom.py`? Is it suggested to leave them to be handled in `azure-cli-core`? @Juliehzl